### PR TITLE
Add a less slow implementation of indexOfObjectWithPredicate

### PR DESCRIPTION
**Pull request by tgoyne**
I don't remember what prompted this, but the commit's been sitting around unmerged for a few months and it seems mildly useful.

### CODE REVIEW

File: Realm/RLMArray.mm [hunk 1]
```diff
@@ -176,6 +176,12 @@ - (void)setValue:(id)value forKey:(NSString *)key {
     [_backingArray setValue:value forKey:key];
 }
 
+- (NSUInteger)indexOfObjectWithPredicate:(__unused NSPredicate *)predicate {
```
Comment (bdash) [inline Realm/RLMArray.mm]: Why __unused?

**Comment by bdash** [comment]
The style guidelines indicate that the opening brace for methods should be on its own line. The new methods in this patch don't follow this, but neither do a lot of the existing code.

:+1:

**Comment by segiddins** [comment]
Seems some RealmSwift tests need to be updated? Otherwise, :+1:

**Comment by bmunkholm** [comment]
The style guideline must be followed. If other parts of the code isn't then it just has to be fixed. (we should have a styleguide linter... )

**Comment by tgoyne** [comment]
I'd prefer to just update the style guidelines to match the codebase over reformatting everything.

**Pull request merged**